dbcon/execplan/calpontsystemcatalog.cpp

```diff
@@ -3607,6 +3615,237 @@ const CalpontSystemCatalog::ROPair CalpontSystemCatalog::tableRID(const TableNam
   throw IDBExcept(ERR_TABLE_NOT_IN_CATALOG, args);
 }
 
+// This function is similar to CalpontSystemCatalog::tableRID, except that
+// instead of returning a ROPair for the table, it returns the OID for the
+// AUX column for the table
+CalpontSystemCatalog::OID CalpontSystemCatalog::tableAUXColumnOID(const TableName& tableName,
+                                                                  int lower_case_table_names)
+{
+  TableName aTableName;
+  aTableName.schema = tableName.schema;
+  aTableName.table = tableName.table;
+  if (lower_case_table_names)
+  {
+    boost::algorithm::to_lower(aTableName.schema);
```

Comment:
IIRC all boost::to_lower() invocations share the same mutex when they access system's locale. What is inside aTableName.schema or aTableName.table when non-ASCII is used? Maybe we can just use ASCII only tolower converter here?

Reply:
I am not sure if using the ASCII only `tolower` would be correct if `aTableName.schema` and `aTableName.table` contain non-ASCII strings. @dhall-MariaDB can probably answer why we use `boost::algorithm::to_lower()` here for converting to lower case.